samples/chaincode/confidential-escrow/chaincode/transactions/digitalAsset.go

```diff
@@ -0,0 +1,700 @@
+			DataType:    "string",
+			Required:    true,
+		},
+		{
+			Tag:         "decimals",
+			Label:       "Decimal Places",
+			Description: "Decimal Places in Digital Asset",
+			DataType:    "number",
+			Required:    true,
+		},
+		{
+			Tag:         "totalSupply",
+			Label:       "Total Supply",
+			Description: "Total Supply of the Digital Asset",
+			DataType:    "number",
+			Required:    true,
+		},
+		{
+			Tag:         "owner",
+			Label:       "Owner Identity",
+			Description: "Identitiy of Digital Asset's creator",
+			DataType:    "string",
+			Required:    true,
+		},
+		{
+			Tag:         "issuedAt",
+			Label:       "Issued At",
+			Description: "Time at which this token was created",
+			DataType:    "datetime",
+			Required:    false,
+		},
+		{
+			Tag:         "issuerHash",
+			Label:       "Issuer Certificate Hash",
+			Description: "Hash of Issuer's Certificate who created this Digital Asset",
+			DataType:    "string",
+			Required:    true,
+		},
+	},
+
+	Routine: func(stub *sw.StubWrapper, req map[string]interface{}) ([]byte, errors.ICCError) {
+		name, _ := req["name"].(string)
+		symbol, _ := req["symbol"].(string)
+		decimals, _ := req["decimals"].(float64)
+		totalSupply, _ := req["totalSupply"].(float64)
+		owner, _ := req["owner"].(string)
+		issuerHash, _ := req["issuerHash"].(string)
+
+		assetMap := make(map[string]interface{})
+		assetMap["@assetType"] = "digitalAsset"
+		assetMap["name"] = name
+		assetMap["symbol"] = symbol
+		assetMap["decimals"] = decimals
+		assetMap["totalSupply"] = totalSupply
+		assetMap["owner"] = owner
+		assetMap["issuedAt"] = time.Now()
+		assetMap["issuerHash"] = issuerHash
+
+		digitalAsset, err := assets.NewAsset(assetMap)
+		if err != nil {
+			return nil, errors.WrapError(err, "Failed to create digital asset")
+		}
+
+		_, err = digitalAsset.PutNew(stub)
+		if err != nil {
+			return nil, errors.WrapErrorWithStatus(err, "Error saving digital asset on blockchain", err.Status())
+		}
+
+		assetJSON, nerr := json.Marshal(digitalAsset)
+		if nerr != nil {
+			return nil, errors.WrapError(nil, "failed to encode asset to JSON format")
+		}
+
+		logMsg, ok := json.Marshal(fmt.Sprintf("New Digital Asset created: %s", name))
+		if ok != nil {
+			return nil, errors.WrapError(nil, "failed to encode asset to JSON format")
+		}
+
+		events.CallEvent(stub, "createDigitalAssetLog", logMsg)
+
+		return assetJSON, nil
+	},
+}
+
+var ReadDigitalAsset = transactions.Transaction{
+	Tag:         "readDigitalAsset",
+	Label:       "Read Digital Asset",
+	Description: "Read a Digital Asset by its symbol",
+	Method:      "GET",
+	Callers: []accesscontrol.Caller{
+		{
+			MSP: "Org1MSP",
+			OU:  "admin",
+		}, {
+			MSP: "Org2MSP",
+			OU:  "admin",
+		},
+	},
+
+	Args: []transactions.Argument{
+		{
+			Tag:         "uuid",
+			Label:       "UUID",
+			Description: "UUID of the Digital Asset to read",
+			DataType:    "string",
+			Required:    true,
+		},
```

Comment:
Instead of receiving the uuid section of the @key of the asset you could receive directly a reference to the asset. The argument could be:
```
		{
			Tag:         "digitalAsset",
			Label:       "Digital Asset",
			Description: "Digital Asset to read",
			DataType:    "->digitalAsset",
			Required:    true,
		},
```

This allows you to call the transaction with two options of body:
```
{
    "@assetType": "digitalAsset",
    "@key": "digitalAsset:[UUID]",
}
```
Or
```
{
    "@assetType": "digitalAsset",
    "symbol": "[TokenSymbol]",
}
```

This would the "CC-Tools" way of doing it.

Reply:
Also, the argument would "arrive" in the transaction routine already validated by CC-Tools.